### DISCUSSION TITLE
Add a Benchmark to Editor Settings

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10151] [General Discussions]
Godot, editor settings, add a benchmark button that runs a few tests with timers and outputs a log of tests ran and time taken.

This will allow users to see if it's worth transferring their entire game project to the latest Godot if the numbers are waaay better.

So it benchmarks things like sprites on the screen, shader performance, particle performance, etc etc.

Just an idea to think about.

**Comment by AThousandShips** [comment]
Would this benchmark the project itself? Or the engine?

**Reply by shak2** [reply]
It would benchmark the engine on the current hardware using a standardised suite of tests that push the system to it's limits and return scores.

**Reply by AThousandShips** [reply]
There's a dedicated repo for that https://github.com/godotengine/godot-benchmarks
